src/fieldtypes/CollectionFieldType.php

```diff
@@ -0,0 +1,148 @@
+     */
+    public function normalizeValue($value, ElementInterface $element = null)
+    {
+        if (is_array($value)) {
+            return $value;
+        }
+        return json_decode($value);
+    }
+
+    // Static Methods
+    // =========================================================================
+
+    /**
+     * Returns the display name of this class.
+     *
+     * @return string The display name of this class.
+     */
+    public static function displayName(): string
+    {
+        //category is the filename inside ./translations/en/
+        return Craft::t('shopify', 'Shopify Collection');
+    }
+
+    /**
+     * returns the template-partial an editor sees when editing plugin-content on a page
+     *
+     * @param mixed $value
+     * @param ElementInterface $element
+     * @return string
+     * @throws \Twig_Error_Loader
+     * @throws \yii\base\Exception
+     */
+    public function getInputHtml($value, ElementInterface $element = null): string
+    {
+        /** @var \shopify\models\Settings $settings */
+        $settings = \shopify\Shopify::getInstance()->getSettings();
+
+        $defaultOptions = [];
+        $options = [];
+
+        // SMART COLLECTIONS:
+        $count = Shopify::getInstance()->service->getSmartCollectionsCount($defaultOptions);
+        $collectionsData = Shopify::getInstance()->service->getCollections(
+            $settings->allSmartCollectionsEndpoint,
+            $defaultOptions
+        );
+
+        if ($smartCollections = $this->_getCollections($count, $collectionsData, 'smart_collections', $defaultOptions)) {
+            $options = array_merge($options, $smartCollections);
+        }
+
+        // CUSTOM COLLECTIONS:
+        $count = Shopify::getInstance()->service->getCustomCollectionsCount($defaultOptions);
+        $collectionsData = Shopify::getInstance()->service->getCollections(
+            $settings->allCustomCollectionsEndpoint,
+            $defaultOptions
+        );
+
+        if ($customCollections = $this->_getCollections($count, $collectionsData, 'custom_collections', $defaultOptions)) {
+            $options = array_merge($options, $customCollections);
+        }
+
+        return Craft::$app->getView()->renderTemplate('shopify/_select', [
+            'filter_class' => $this->handle . '_filter',
+            'selected_only_class' => $this->handle . '_selected_only',
+            'clear_selected_class' => $this->handle . '_clear_selected',
+            'wrapper_class' => $this->handle . '_wrapper',
+            'name' => $this->handle,
+            'value' => $value,
+            'field' => $this,
+            'options' => $options,
+            'type' => 'collections',
+        ]);
+    }
+
+    /**
+     * @param int $count
+     * @param array $collectionsData
+     * @param string $key
+     * @param array $defaultOptions
+     * @return array
+     */
+    private function _getCollections($count, $collectionsData, $key, $defaultOptions)
```

Comment:
Shouldn't these be `public` so the user can make their own field in the twig templates as mentioned in the upstream documentation?

Reply:
The private functions are only helper functions specific to that class. That was me OOP-ing stuff. The public functions in this class remain as they were.

Reply:
Ah, I see. I don't know how to read code.